src/shared/parameter_v2/scripts/config_yaml_loader.py

```diff
@@ -0,0 +1,297 @@
+        """Loads the yamls into an dictionary. Any errors while in the yaml
+        syntax will raise to the main thread. We also adjust how the dictionary
+        is stored for easier access later.
+
+        :raises: ConfigYamlMalformed
+        :param yaml_paths: the path to all the config yamls
+        :type yaml_paths: list of str
+        :returns: config_medata dict representing the data to generate
+        :rtype: dict
+
+        """
+        raw_config_metadata = {}
+
+        for filename in yaml_paths:
+            with open(filename, "r") as param_yaml:
+
+                try:
+                    # extract config name from filename
+                    _, tail = os.path.split(filename)
+
+                    # safe load yaml into dictionary
+                    raw_config_metadata[tail] = list(yaml.safe_load_all(param_yaml))
+
+                    if len(raw_config_metadata[tail]) == 1:
+
+                        # include only in file
+                        if isinstance(raw_config_metadata[tail][0], dict):
+                            raw_config_metadata[tail] = {
+                                INCLUDE_KEY: raw_config_metadata[tail][0][INCLUDE_KEY]
+                            }
+
+                        # parameter definitions only in file
+                        if isinstance(raw_config_metadata[tail][0], list):
+                            raw_config_metadata[tail] = {
+                                PARAMETER_KEY: raw_config_metadata[tail][0]
+                            }
+
+                    elif len(raw_config_metadata[tail]) == 2:
+
+                        # include and param definition in file
+                        raw_config_metadata[tail] = {
+                            INCLUDE_KEY: raw_config_metadata[tail][0][INCLUDE_KEY],
+                            PARAMETER_KEY: raw_config_metadata[tail][1],
+                        }
+
+                    else:
+                        raise ConfigYamlMalformed(
+                            "More than two yaml documents in {}".format(tail)
+                        )
+
+                except yaml.YAMLError as ymle:
+                    raise ConfigYamlMalformed(
+                        "Check malformed {} \n {}".format(tail, ymle)
+                    ) from None
+
+        return raw_config_metadata
+
+    @staticmethod
+    def __validate_config_metadata(config_metadata: dict):
+        """Validates the config_metadata that was loaded against the
+        dynamic_parameter_schemas and then checks for duplicate includes
+        and duplicate parameters in the same config.
+
+        :raises: ConfigYamlMalformed
+        :raises: ConfigSchemaViolation
+        :param config_metadata: Metadata describing params and config includes
+        :type config_metadata: dict
+
+        """
+
+        for config_file, metadata in config_metadata.items():
+
+            if INCLUDE_KEY in metadata:
+
+                # check schema
```

Comment:
Nit: "Check schema" is a little vague. What are we checking for? Perhaps "Validate correct schema format" would work.

Reply:
changed